servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/MessageWithDeliveryTag.java

```diff
@@ -5,21 +5,30 @@
 
 import org.apache.qpid.proton.message.Message;
 
+import java.util.Arrays;
+
 public class MessageWithDeliveryTag {
     private final Message message;
     private final byte[] deliveryTag;
 
-    public MessageWithDeliveryTag(Message message, byte[] deliveryTag)
-    {
+    public MessageWithDeliveryTag(Message message, byte[] deliveryTag) {
         this.message = message;
-        this.deliveryTag = deliveryTag;
+
+        if (deliveryTag == null) {
+            this.deliveryTag = null;
+        } else {
+            this.deliveryTag = Arrays.copyOf(deliveryTag, deliveryTag.length);
+        }
     }
 
     public Message getMessage() {
         return message;
     }
 
     public byte[] getDeliveryTag() {
-        return deliveryTag;
+        if (this.deliveryTag == null) {
```

Comment:
[SpotBugs-P2]: .getDeliveryTag() may expose internal representation by returning MessageWithDeliveryTag.deliveryTag

Returning a reference to a mutable object value stored in one of the object's fields exposes the internal representation of the object.  If instances are accessed by untrusted code, and unchecked changes to the mutable object would compromise security or other important properties, you will need to do something different. Returning a new copy of the object is better approach in many situations.